### DISCUSSION TITLE
Making graph take up entire card div

### DISCUSSION THREAD

**Discussion by rahulbrahmal** [reactivemarkets/react-financial-charts#559] [Q&A] [answered]
Hi!

I'm currently using react-financial-charts in a project of mine and am running into a slight problem. On a specific page, I have cards which resize dynamically based on the size of the container (I'm using the GRID and BOX components in chakra UI to do this). I would like a graph in these, but noticed that the graph does not extend to the entire area of the div. I know that with the `withSize` HOC that I can specify a minWidth, however, this does not resize when the card shrinks (for example when the page size is made smaller.

I've attached the two snippets of code below as well. Would anyone have a good way of potentially doing this?

`
 <Box>
          <LineGraph
            data={data.data}
            xLabel={data.xLabel}
            yLabel={data.yLabel}
          />
 </Box>
`

`
export default withSize({
  style: {
    minHeight: 355
  },
// @ts-ignore
})(withDeviceRatio()(LineGraph));
`

<img width="892" alt="image" src="https://user-images.githubusercontent.com/24690504/128819023-1aa3e4e8-42a3-4891-a0a1-d66dac1a5b36.png">

**Comment by markmcdowell** [comment]
You just need to stretch the outer container to fit the space, you don't need a min height. Something like `flex: 1 1 auto` on your Box.

**Comment by dev-nll** [comment, marked as answer]
You might want to try to wrap everything in Flex (also part of Chakra), set
it to row mode, and specify flex="1" inside the Box props

**Reply by rahulbrahmal** [reply]
Thank you so much! @dev-nll 's response worked perfectly! One additional thing I did need to do was set the Box props width and height to '100%'


```
<Flex direction="column">
  <Box flex="1" width="100%" height="100%">
    <LineGraph />
  </Box>
</Flex>

```
